utils/vcsdata.go

```diff
@@ -73,75 +75,49 @@ func (this *VcsDetails) GetVcsData(path string) (revision, refUrl string, err er
 		subPaths = append(subPaths, subPath)
 	}
 	if size := len(subPaths); size > 0 {
-		this.healthCheack()
+		this.clearCacheIfExceedsMax()
 		for _, v := range subPaths {
-			this.vcsDirectory.Store(v, vcsDataResult)
+			this.vcsDir.Store(v, vcsDetailsResult)
 		}
-		this.increment(int32(size))
+		this.incCacheSize(int32(size))
 	}
 	return
 }
 
-func (this *VcsDetails) healthCheack() {
-	if this.get() > MAX_ENTRIES {
-		this.vcsDirectory = sync.Map{}
-		this.vcsDirectorySize = new(int32)
+func (this *VcsCache) clearCacheIfExceedsMax() {
+	if this.getCacheSize() > MAX_ENTRIES {
+		this.vcsDir = sync.Map{}
+		this.vcsDirSize = new(int32)
 	}
 }
 
-func tryGetGitDetails(path string, this *VcsDetails) (string, string, error) {
-	dotGitPath := filepath.Join(path, ".git")
-	exists, err := fileutils.IsDirExists(dotGitPath, false)
+func tryGetGitDetails(path string, this *VcsCache) (string, string, error) {
+	exists, err := fileutils.IsDirExists(filepath.Join(path, ".git"), false)
 	if exists {
-		return extractGitInfo(dotGitPath)
+		return extractGitDetails(path)
 	}
 	return "", "", err
 }
 
-func extractGitInfo(path string) (revision string, refUrl string, err error) {
-	dotGitPath := filepath.Join(path, "HEAD")
-	file, er := os.Open(dotGitPath)
-	if er != nil {
-		err = er
-		return
+func extractGitDetails(path string) (string, string, error) {
+	gitService := NewGitManager(path)
+	if err := gitService.ReadConfig(); err != nil {
+		return "", "", err
 	}
-	defer file.Close()
-	scanner := bufio.NewScanner(file)
-	for scanner.Scan() {
-		text := scanner.Text()
-		if strings.HasPrefix(text, "ref") {
-			refUrl = strings.TrimSpace(strings.SplitAfter(text, ":")[1])
-			break
+	return gitService.GetRevision(), gitService.GetUrl(), nil
+
+}
+
+func (this *VcsCache) searchCache(path string) (*vcsDetails, bool) {
```

Comment:
The bool return value by this function is probably not needed, because if the git details ate not found, the function can return nil.

Reply:
No, we may have subpath with nil as a value, which means that we already searched there and didn't find any .git.